modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -545,6 +545,85 @@ ADLAROMANCEStressUpdateBase::computeResidual(const ADReal & effective_trial_stre
   return total_rom_effective_strain_inc - scalar;
 }
 
+Real
+ADLAROMANCEStressUpdateBase::computeCreepStrainRate(const ADReal & effective_trial_stress)
+{
+  // Update new stress
+  auto trial_stress_mpa = effective_trial_stress * 1.0e-6;
+  ADReal dtrial_stress_dscalar = 0.0;
+
+  // Update stress if strain is being applied, i.e. non-testing simulation
+  _input_values[_stress_input_index] = trial_stress_mpa;
+
+  // Update weights with new stress
+  _weights = _non_stress_weights;
+  computeTileWeight(_weights, _input_values[_stress_input_index], _stress_input_index);
+  auto dweights_dstress = _non_stress_weights;
+  computeTileWeight(
+      dweights_dstress, _input_values[_stress_input_index], _stress_input_index, true);
+
+  // Check window with new stress
+  checkInputWindow(_input_values[_stress_input_index],
+                   _window_failure[_stress_input_index],
+                   _global_limits[_stress_input_index]);
+
+  // Save extrapolation as a material property in order quantify adequate tiling range
+  _extrapolation[_qp] = 0.0;
+  for (unsigned int t = 0; t < _num_tiles; ++t)
+    _extrapolation[_qp] += MetaPhysicL::raw_value(_weights[t]);
+
+  ADReal total_rom_effective_strain_inc = 0.0;
+  ADReal dtotal_rom_effective_strain_inc_dstress = 0.0;
+
+  // Run ROM if all values are within windows.
+  for (unsigned int t = 0; t < _num_tiles; ++t)
+  {
+    if (_weights[t])
+    {
+      const ADReal rom = computeROM(t, _strain_output_index);
+      total_rom_effective_strain_inc += _weights[t] * rom;
+      dtotal_rom_effective_strain_inc_dstress +=
+          _weights[t] * computeROM(t, _strain_output_index, true) + dweights_dstress[t] * rom;
+    }
+  }
+
+  Real creep_rate = MetaPhysicL::raw_value(total_rom_effective_strain_inc) / _dt;
+
+  return creep_rate;
+}
+
```

Comment:
@lynnmunday @sapitts Is this a good way to get the creep strain rate in LAROMANCE given the effective stress? For models that behave according to a simple relation like a regular power law is straightforward, but I suspect I am not doing this right for LAROMANCE.

Reply:
Hi Antonio! sorry for the delay in responding. The way that you're using the stress and strain index variables looks right to me. And you've accounted for the LaRomance models assumption of stress units of MPa (not Pa like most of our models). I'm not sure that all of the tiling logic is duplicated over faithfully, but I haven't worked as much with the tiling as Lynn has.

Is there a way to reuse the calculation from the "regular" strain calculation when called by the return mapping algorithm call, since the LaRomance calculation can be expensive because of the number of polynomial calculations involved? There is the material property `_creep_rate` that might be useful?